src/ess/reflectometry/tools.py

```diff
@@ -221,10 +271,15 @@ def cost(scaling_factors):
 
     sol = opt.minimize(cost, [1.0] * (len(curves) - 1))
     scaling_factors = (1.0, *map(float, sol.x))
-    return [
-        scaling_factor * curve
-        for scaling_factor, curve in zip(scaling_factors, curves, strict=True)
-    ], scaling_factors
+
+    wfc[ScalingFactorForOverlap[SampleRun]] = dict(
+        zip(curves.keys(), scaling_factors, strict=True)
+    )
+
+    if CriticalEdgeKey in wfc.keys():
```

Comment:
I'm not super happy about the implementation with the `CriticalEdgeKey`...
Suggestions welcome.